cartography/intel/aws/cloudtrail_management_events.py

```
@@ -0,0 +1,472 @@
+    logger.info(
+        f"Retrieved {len(all_events)} total role assumption events from region '{region}'"
+    )
+
+    return all_events
+
+
+@timeit
+def transform_cloudtrail_events_to_role_assumptions(
+    events: List[Dict[str, Any]],
+    region: str,
+    current_aws_account_id: str,
+) -> List[Dict[str, Any]]:
+    """
+    Transform raw CloudTrail events into aggregated role assumption relationships.
+
+    This function performs the complete transformation pipeline:
+    1. Extract role assumption events from CloudTrail data
+    2. Aggregate events by (source_principal, destination_principal) pairs
+    3. Return aggregated relationships ready for loading
+
+    :type events: List[Dict[str, Any]]
+    :param events: List of raw CloudTrail events from lookup_events API
+    :type region: str
+    :param region: The AWS region where events were retrieved from
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :rtype: List[Dict[str, Any]]
+    :return: List of aggregated role assumption relationships ready for loading
+    """
+    role_assumptions = []
+
+    logger.info(
+        f"Transforming {len(events)} CloudTrail events to role assumptions for region '{region}'"
+    )
+
+    for event in events:
+        # Extract role assumption details from CloudTrail event
+        event_name = event.get("EventName")
+        if not event_name:
+            logger.debug(
+                f"Event missing EventName field: {event.get('EventId', 'unknown')}"
+            )
+            continue
+
+        # Parse CloudTrail event JSON once
+        cloudtrail_event = None
+        if event.get("CloudTrailEvent"):
+            try:
+                cloudtrail_event = json.loads(event["CloudTrailEvent"])
+            except (json.JSONDecodeError, TypeError):
+                logger.debug(
+                    f"Failed to parse CloudTrail JSON for event {event.get('EventId', 'unknown')}"
+                )
```

For a first PR on this, let's not apply this level of defensive programming yet. Let the exception fail.